nncf/experimental/torch/nas/bootstrapNAS/search/supernet.py

```diff
@@ -0,0 +1,166 @@
+
+
+class SuperNetwork:
+    """
+    An interface for handling pre-trained super-networks. This class can be used to quickly implement
+    third party solutions for subnetwork search on existing super-networks.
+    """
+
+    def __init__(self, elastic_ctrl: ElasticityController, nncf_network: NNCFNetwork):
+        """
+        Initializes the super-network interface.
+
+        :param elastic_ctrl: Elasticity controller to activate subnetworks
+        :param nncf_network: NNCFNetwork that wraps the original PyTorch model.
+        """
+        self._m_handler = elastic_ctrl.multi_elasticity_handler
+        self._elasticity_ctrl = elastic_ctrl
+        self._model = nncf_network
+
+    @classmethod
+    def from_checkpoint(
+        cls,
+        model: TModel,
+        nncf_config: NNCFConfig,
+        supernet_elasticity_path: str,
+        supernet_weights_path: str,
+    ) -> "SuperNetwork":
+        """
+        Loads existing super-network weights and elasticity information, and creates the SuperNetwork interface.
+
+        :param model: base model that was used to create the super-network.
+        :param nncf_config: configuration used to create the super-network.
+        :param supernet_elasticity_path: path to file containing state information about the super-network.
+        :param supernet_weights_path: trained weights to resume the super-network.
+        :return: SuperNetwork with wrapped functionality.
+        """
+        nncf_network = create_nncf_network(model, nncf_config)
+        compression_state = torch.load(supernet_elasticity_path, map_location=torch.device(nncf_config.device))
+        model, elasticity_ctrl = resume_compression_from_state(nncf_network, compression_state)
+        model_weights = torch.load(supernet_weights_path, map_location=torch.device(nncf_config.device))
+        load_state(model, model_weights, is_resume=True)
+        elasticity_ctrl.multi_elasticity_handler.activate_maximum_subnet()
+        return SuperNetwork(elasticity_ctrl, model)
+
+    def get_search_space(self) -> Dict:
+        """
+        :return: dictionary with possible values for elastic configurations.
+        """
+        return self._m_handler.get_search_space()
+
+    def get_design_vars_info(self) -> Tuple[int, List[int]]:
+        """
+        :return: number of possible values in subnet configurations and
+        the number of possible values for each elastic property.
+        """
+        self._m_handler.get_design_vars_info()
+
+    def eval_subnet_with_design_vars(self, design_config: List, eval_fn: ValFnType, **kwargs) -> Any:
+        """
+
+        :returns the value produced by the user's function to evaluate the subnetwork.
+        """
+        self._m_handler.activate_subnet_for_config(self._m_handler.get_config_from_pymoo(design_config))
+        return eval_fn(self._model, **kwargs)
+
+    def eval_active_subnet(self, eval_fn: ValFnType, **kwargs) -> Any:
+        """
+
```

Comment:
```suggestion
```

Reply:
Here and below, please remove empty line.
It's needed between method description and params only, because you have just `:param:` and `:return:` (which is ok in obvious cases) 
![image](https://github.com/openvinotoolkit/nncf/assets/4014476/12f2e817-ca8a-4017-9a91-7c5ff3206c1b)

Reply:
@ljaljushkin  Fixed. Thanks!